lib/Stampie/Mailer.php

```diff
@@ -173,4 +200,54 @@ protected function buildIdentityString($identities)
     {
         return IdentityUtils::buildIdentityString($identities);
     }
+
+    /**
+     * Take a Message and return a Stampie Response
+     *
+     * @param MessageInterface $message
+     *
+     * @return Response
+     */
+    private function doSend(MessageInterface $message)
+    {
+        $content = $this->format($message);
+        $headers = $this->getHeaders();
+        $files = $this->getFiles($message);
+
+        if (!empty($files)) {
+            // HTTP query content
+            parse_str($content, $fields);
+            $builder = new MultipartStreamBuilder();
+
+            foreach ($fields as $name => $contents) {
+                if (!is_array($contents)) {
+                    $builder->addResource($name, $contents);
+                } else {
+                    foreach ($contents as $c) {
+                        $builder->addResource($name.'[]', $c);
+                    }
+                }
+            }
+
+            // Add files to request
+            foreach ($files as $key => $items) {
+                foreach ($items as $name => $path) {
+                    $options = [];
+                    if (!is_numeric($name)) {
+                        $options['filename'] = $name;
+                    }
+                    $contents = fopen($path, 'r');
+                    $builder->addResource($key, $contents, $options);
+                }
+            }
+
+            $content = $builder->build();
+            $headers['Content-Type'] = 'multipart/form-data; boundary='.$builder->getBoundary();
```

Comment:
@henrikbjorn if the mailer returns a non-empty array in ``getFiles``, it means they want a multipart submission. Mailers not using multipart submission are handling attachments in the ``format`` method instead, as they are part of the main content (see the Mandrill implementation for instance)